components/Article/withReadingProgress.js

```diff
@@ -220,14 +220,14 @@ const withReadingProgress = WrappedComponent => {
             progressElementIndex: nextIndex
           })
           return {
-            nodeId: progressElement && progressElement.id,
+            nodeId: progressElement && progressElement.dataset && progressElement.dataset.pos,
```

Comment:
use `getAttribute('data-pos')` instead, needs no extra check for `dataset` presence